### PR TITLE
Bug-Fix issue: 182 Picker in AddMaintainanceView is not working

### LINKED ISSUE #182: BUG - Picker in AddMaintenanceView is not working
## Describe the bug
It is not possible to select another vehicle in the AddMaintenanceView.

## To Reproduce
1. Add more than on vehicle in the settings screen
2. Add a new maintenance
3. Try to select the second vehicle in the picker

## Additional Context
* Problem is that the picker selection should only use id not the vehicle object.

@mikaelacaron  I already have a working bugfix, but I wanted to give someone else the chance to fix it. If the solution is needed immediately I can open a PR with the fix.

### PR DESCRIPTION
**Pull request by rramirez-dev**
Updated the vehicle picker to user the vehicle id and not the vehicle object

# What it Does
* Closes #182 
* Describe what your change does
   The fix allows the user to select a vehicle from the 
   vehicle picker in the AddMaintainanceView form that
   is not the first vehicle in the list.

# How I Tested
* Add a list of steps to show the functionality of your feature
For example:
* Run the application
* Tap the (+) button
* See this screen
etc...

1. Launch the App
2. Load the dashboard by navigating to a different tab     and then select the dashboard tab again.
3. Tap the plus sign to add a new event
4. Enter a title
5. Select a vehicle that is not the first one on the list
6. Tap Add

# Notes
* Anything else that should be noted about how you implemented this feature?
* I filtered the vehicles array based on the id of the vehicle that the user selected. I then get the first item in the array and pass it to the MaintainanceEvent view.

# Screenshot
Simulator Screen Recording - iPhone 15 - 2023-10-20 at 21 49 41

### CODE REVIEW

**Review by mikaelacaron — COMMENTED** [review]
Can you make this work using the `Vehicle` object as opposed to the id? if not use the id and change the variable to `selectedVehicleID`
Also there's some indentation that's incorrect, your Xcode should be set to `4` spaces

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
nevermind sorry for the confusion, go ahead and use the `id`, that's how it's done in the `AddOdometerReadingView`

Also note you don't need the `vehicle as Vehicle` anymore because of that, and technically with the current way that you fixed this the `if !vehicles.isEmpty` check wouldn't be needed, because you made the `selectedVehicle` optional

**Comment by rramirez-dev** [comment]
So I should remove "vehicle as Vehicle" from the tag and replace it with "vehicle.id".

**Comment by mikaelacaron** [comment]
> So I should remove `vehicle as Vehicle` from the tag and replace it with `vehicle.id`

Yes because it's not necessary to cast the vehicle as it's own type, the goal is to use the `id` as the main point of data for the picker, like you had it before

File: Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift [hunk 1]
```diff
@@ -34,9 +34,9 @@ struct AddMaintenanceView: View {
                 }
                 
                 Section {
-                    Picker(selection: $selectedVehicle) {
-                        ForEach(vehicles) { vehicle in
-                            Text(vehicle.name).tag(vehicle as Vehicle)
+                    Picker(selection: $selectedVehicleID) {
+                        ForEach(vehicles, id: \.self) { vehicle in
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift]:
```suggestion
                        ForEach(vehicles) { vehicle in
```

This is not needed, it's about if the elements in the `ForEach` are `Identifiable`, which they are, the `id` doesn't need to be explicitly noted

File: Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift [hunk 2]
```diff
@@ -67,23 +67,21 @@ struct AddMaintenanceView: View {
...
-                                                         notes: notes,
-                                                         vehicle: selectedVehicle)
-                            addTapped(event)
+                        if let selectedVehicleID {
+                            if let vehicle = vehicles.filter({ $0.id == selectedVehicleID }).first {
+                                let event = MaintenanceEvent(title: 
+                                                            title,
+                                                            date: date,
+                                                            notes: notes,
+                                                            vehicle: vehicle)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift]: Going to push a change to fix the alignment of this

**Review by mikaelacaron — APPROVED** [review]
Good work! Just a small comment and I can merge this

**Pull request merged**